framework/web/Response.php

```diff
@@ -1061,10 +1061,14 @@ protected function defaultFormatters()
      * Prepares for sending the response.
      * The default implementation will convert [[data]] into [[content]] and set headers accordingly.
      * @throws InvalidConfigException if the formatter for the specified format is invalid or [[format]] is not supported
+     *
+     * @see https://tools.ietf.org/html/rfc7231#page-53
+     * @see https://tools.ietf.org/html/rfc7232#page-18
      */
     protected function prepare()
     {
-        if ($this->statusCode === 204) {
+        if ($this->statusCode === 204 || $this->statusCode === 304) {
```

Comment:
Isn't it the same for 100, 101, 102, 205?

Reply:
The specification for 1xx does not say anything about of the content body response, it is not explicit like  rfc7231/rfc7232 are. By other hand the specification for 205 code status is more complex ([see here](https://tools.ietf.org/html/rfc7231#page-53)), maybe we need to prepare that response according to the rfc7231, but I think it is better to break it out in other issue.

Reply:
OK.